Translation/Translation/Transliteration.cs

```diff
@@ -0,0 +1,105 @@
+﻿
+namespace Translation
+{
+    public class Transliteration
+    {
+        const string RUSSIAN = "абвгдеёжзийклмнопрстуфхцчшщъыьэюя";
+        const string LATIN = " abvgdeyozhiklmnprstufhc";
+        public Transliteration() { }
+
+        // This method checks the string for validity
+        public void CheckString(string str)
+        {
+            str = str.ToLower();
+            string russianСonsonants = "бвгджзклмнпрстфхцчшщ";
+            int counterRussianSymbols = 0;
+            int counterLatinSymbols = 0;
+
+            if (str.Length == 0 || str == null)
+            {
+                throw new StringException("Empty String. Try again");
+            }
+
+
+            foreach (char symbol in str)
+            {
+                if (RUSSIAN.Contains(symbol) == false && LATIN.Contains(symbol) == false)
+                {
+                    throw new StringException("Invalid characters. Try again");
+                }
+                if (RUSSIAN.Contains(symbol))
+                {
+                    if (str[0] == 'ь' || str[0] == 'ъ') // 'ь' and 'ъ' cannot be at the beginning of the string
+                    {
+                        throw new StringException("Invalid characters. Try again");
+                    }
+                    for (int i = 1; i < str.Length; i++)
+                    {
+                        if ((str[i] == 'ь' && russianСonsonants.Contains(str[i - 1]) == false) || (str[i] == 'ъ' && russianСonsonants.Contains(str[i - 1]) == false)) // before 'ь' and 'ъ' can be only consonants
+                        {
+                            throw new StringException("Invalid characters. Try again");
+                        }
+                    }
+                    counterRussianSymbols++;
+                }
+                if (LATIN.Contains(symbol))
+                {
+                    for (int i = 0; i < str.Length; i++)
+                    {
+                        if(str[i] == 'c' && str[i + 1] != 'h') // ч - ch, 'c' is not used anywhere 
+                        {
+                            throw new StringException("Invalid characters. Try again");
+                        }
+                    }
+                        counterLatinSymbols++;
+                }
+                if (counterRussianSymbols > 0 && counterLatinSymbols > 0)
+                {
+                    throw new StringException("Invalid characters. Try again");
+                }
+            }
+        }
+
+        //This method translates the russian string to latin and vice versa
+        public string TranslateString(string str)
+        {
```

Comment:
Make these massives static readonly, not to createt them each time you call method. If you call method 1K times we will create 4K massives, what for?